src/i18n/scripts/intl-imports.js

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env node
+const fs = require('fs/promises');
+
+const i18nDir = 'src/i18n';
+const messagesDir = `${i18nDir}/messages`;
+
+async function generateDirectoryIndex(directory) {
+    console.log('generateDirectoryIndex', directory);
+
+    const importLines = [];
+
+    const messagesLines = [];
+
+    try {
+        const files = await fs.readdir(`${messagesDir}/${directory}`, { withFileTypes: true });
+
+        const jsonFiles = files.filter(file => file.isFile() && file.name.endsWith('.json'))
+
+        if (!jsonFiles.length) {
+            console.error(`Skipping ${directory} because no .json translation files were found.`);
+            return;
+        }
+
+        for (const [i, file] of jsonFiles.entries()) {
+            const languageCode = file.name.replace(/\.json$/, '');
+            const languageCodeDash = languageCode.toLowerCase().replace(/_/g, '-')
+            const uniqueMessageVariable = `messages${i}${languageCode}`;
+            const filePath = `${messagesDir}/${directory}/${languageCode}.json`
+
+            try {
+                const entries = JSON.parse(await fs.readFile(filePath, { encoding: 'utf8' }));
+
+                if (!Object.keys(entries).length) {
+                    console.error(`Skipping ${directory}'s ${languageCode} due to empty translation files were found.`);
```

Comment:
The error message as is implies to me that finding any empty translation file in the directory would lead to skipping a language in that directory. I think this suggested wording should make it more clear that we only want to skip a language when that specific language file is empty.
```suggestion
                    console.error(`Skipping ${directory}'s ${languageCode} due to empty translation file.`);
```